single_kernel_mongo/events/backups.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Manager for handling backup events."""
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from ops import MaintenanceStatus
+from ops.charm import ActionEvent, RelationJoinedEvent
+from ops.framework import Object
+
+from single_kernel_mongo.config.relations import ExternalRequirerRelations
+from single_kernel_mongo.exceptions import (
+    InvalidArgumentForActionError,
+    InvalidPBMStatusError,
+    ListBackupError,
+    PBMBusyError,
+    RestoreError,
+    ResyncError,
+    SetPBMConfigError,
+    WorkloadExecError,
+    WorkloadServiceError,
+)
+from single_kernel_mongo.lib.charms.data_platform_libs.v0.s3 import (
+    CredentialsChangedEvent,
+    S3Requirer,
+)
+from single_kernel_mongo.utils.event_helpers import (
+    defer_event_with_info_log,
+    fail_action_with_error_log,
+    success_action_with_info_log,
+)
+
+if TYPE_CHECKING:
+    from single_kernel_mongo.abstract_charm import AbstractMongoCharm
```

Comment:
I'm still trying to figure out if it can be removed entirely but I'm not sure.
Note that those are not circular dependencies, but only for type checking which means at run time this is not imported and should  be an issue.
This happens sadly every time you provide yourself (as a class) to one of your attributes if you want to type check correctly.
I still have one idea but I'm ensure it will be more readable than that.